### ISSUE TITLE
Windows 10 Support

### ISSUE THREAD

**Issue by 2agr6v9u7mxu** [alraban/org-recoll#7]
Hi,

I have org-recoll up and running on my Linux machine, however, my Windows 10 machine does not display the results.

I can search the database and a result is returned indicating the number of items found, however, the results underneath are not displayed. 

Are there any assumed Linux dependencies I need to install on Windows (I have gnu utils installed)?

Thanks.

**Comment by alraban** [comment]
I don't have a windows machine to test on, so I can't personally troubleshoot this, but I'm willing to help.

The package uses one basic unix utility (tail), but otherwise has no explicit external dependencies other than recoll. The way the package works is that it calls out to the shell using emacs's built-in shell functions and then parses what the shell returns.  It works well with bash or sh, but obviously those aren't natively available on windows.  I don't know exactly how the built-in shell commands work on windows, but the documentation suggests that emacs would call out to cmd.exe or maybe powershell (unless you've customized the shell environment variable in emacs).  

So first I'd suggest figuring out what shell your emacs is using.  If you have the bash shell available (through gnu-utils), but haven't yet setup emacs to use it that might be an easy fix.  Some relevant documentation: https://www.gnu.org/software/emacs/manual/html_node/efaq-w32/Using-shell.html

If you only have native windows shells the first step in troubleshooting would be to try using the default recoll command-line invocation on the windows shell outside of emacs and see what it returns.  This package works by sending the following default command line invocation to the shell: `recoll -t -A`  It then concatenates your query with that invocation.  So if you can post the output of `recoll -t -A your-query-here` from the windows shell, I can potentially offer some additional advice.

**Comment by 2agr6v9u7mxu** [comment]
Thanks a lot for getting back to me so quickly.

The gnu-utils are in my PATH, however, I was trying to stay away from setting up a full bash environment as it is a work laptop. The first error was complaining about not having tail, however, gnu-utils solved this problem. I tried messing with the recoll on the command line via powershell and it worked as expected using recollq. I tried mixing the commands up within emacs, I just could not get it to output, not even a useful error. 

I do think you have highlighted the issue though, the shell environment is not working correctly within Windows; it's just how to set that up to work without installing a full bash environment. 

Thanks.

**Comment by alraban** [comment]
Hmmm.  If it works in powershell, here are a few thoughts.   

If you invoke M-x shell inside emacs, emacs will give you the shell it's using to call out (i.e. whatever is specified in the SHELL environment variable).  If you use M-x shell and then run recollq or `recoll -t` from the shell inside emacs does it work as expected?  If it doesn't work correctly can you see if emacs is using cmd.exe rather than powershell?  If so, you might be able to set SHELL to use powershell instead which may solve it.

Also, just to rule it out, org-recoll specifically uses `recoll -t` rather than recollq because recollq isn't always available and `recoll -t` is supposed to work the same way (but has some subtle differences).  If you didn't check the output of `recoll -t` specifically in powershell that would be worth doing in case recollq is doing something different.  If you find that recollq behaves better, you canchange org-recoll to use recollq instead by customizing the `org-recoll-command-invocation` variable.

Something else worth testing is adding the tail pipeline to the powershell command and seeing if it still works, i.e. `recoll -t -A -q your-query | tail +3`  If that throws an error that would suggest that powershell doesn't support pipes in the same way that bash does.

Let me know if that turns up anything fruitful.

**Comment by 2agr6v9u7mxu** [comment]
Thank you for sharing your diagnostics step I have managed to fix the problem. When piping the command to tail within cmd.exe I got 2 error, I ended up having to install libintl and libiconv, both available from GNU and once installed the package now runs as expected. 

Side note: Recoll for windows requires manual setup to parse org files, once this setup the entire suite acts like it does on Linux. 

Hopefully these steps will help other people out in the future.

Thanks again!
